### PR TITLE
Fix handleLayoutChange to update dragmode

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?
Fixes a bug introduced in https://github.com/mlflow/mlflow/pull/2408, namely modifying  `handleLayoutChange` in `MetricsPlotPanel` to update `dragmode` of the plot.

Before (`dragmode` can't be changed)
relayout-before

After (`dragmode`can be changed.)
relayout-after

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by smurching — APPROVED** [review]
Thanks @harupy for the fix, looks good! Do you know if there are any other plotly layout fields that we might inadvertently be excluding here / should make sure to include?

**Comment by smurching** [comment]
Looking at the [Plotly API docs](https://plot.ly/javascript/reference/), it seems there are a number of fields nested under "layout" (fields with parent "layout"). Thus I wonder if the right approach here is to:
* Include `newLayout` in `mergedLayout`
* Exclude/convert fields from `newLayout` that aren't valid, e.g. convert `newLayout['xaxis.range[0]']` and `newLayout['xaxis.range[0]']` into an actual range array, as the current logic already does.

Let me know if that makes sense, thanks!

**Comment by harupy** [comment]
Yes, that makes sense. I will fix the code.

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 1]
```diff
@@ -220,12 +221,19 @@ export class MetricsPlotPanel extends React.Component {
...
+      "yaxis.range[1]": newYRange1,
+      "xaxis.autorange": xAxisAutorange,
+      "yaxis.autorange": yAxisAutorange,
+      ...restFields
+    } = newLayout;
+
     let mergedLayout = {
       ...this.state.layout,
+      ...restFields,
     };
```
Comment (harupy) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: @smurching 
I hope this is what you meant in https://github.com/mlflow/mlflow/pull/2459#issuecomment-589999741

Comment (smurching) [reply mlflow/server/js/src/components/MetricsPlotPanel.js]: @harupy yeah looks good!

**Comment by smurching** [comment]
@harupy thanks for the updates - pushed an update to properly handle changes in the `showspikes` property used to control whether spike lines appear (which unfortunately also must be parsed out of the `newLayout`)

**Comment by smurching** [comment]
Would appreciate your review of my changes, otherwise LGTM!

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 2]
```diff
@@ -220,46 +221,58 @@ export class MetricsPlotPanel extends React.Component {
...
-          range: [newXRange0, newXRange1],
-        },
-      };
+      newXAxis.range = [newXRange0, newXRange1];
+      newXAxis.autorange = false;
+    }
+    if (xAxisShowSpikes) {
+      newXAxis.showspikes = true;
+    }
+    if (xAxisAutorange === true) {
```
Comment (harupy) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: Is `=== true` necessary?

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 3]
```diff
@@ -220,46 +221,58 @@ export class MetricsPlotPanel extends React.Component {
...
-    if (newLayout["xaxis.autorange"] === true) {
-      mergedLayout = {
-        ...mergedLayout,
-        xaxis: { autorange: true },
-      };
+    if (yAxisShowSpikes) {
+      newYAxis.showspikes = true;
     }
-    if (newLayout["yaxis.autorange"] === true) {
+    if (yAxisAutorange === true) {
```
Comment (harupy) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: same here

**Comment by harupy** [comment]
@smurching Just small comments otherwise LGTM!

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 4]
```diff
@@ -220,46 +221,58 @@ export class MetricsPlotPanel extends React.Component {
...
-          range: [newXRange0, newXRange1],
-        },
-      };
+      newXAxis.range = [newXRange0, newXRange1];
+      newXAxis.autorange = false;
+    }
+    if (xAxisShowSpikes) {
+      newXAxis.showspikes = true;
+    }
+    if (xAxisAutorange) {
```
Comment (harupy) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: @smurching Thanks for the fix!

**Pull request merged**